File: sterilization-veterinarian/src/main/java/com/esprit/microservice/entities/Pet.java
package com.esprit.microservice.entities;

import lombok.Data;
import lombok.Getter;
import lombok.NonNull;
import lombok.Setter;
import lombok.ToString;

import javax.persistence.*;
import java.util.Date;

@Data
public class Pet {
    
    private String _id, name, type, race ;
    private int age;
    private Date captureDate;

    public Pet() {
    }
}

File: pet/src/model/pet.js
const mongoose = require("mongoose");

const petSchema = mongoose.Schema(
    {
        name: {
            type: String,
            required: true,
        },
        type: {
            type: String,
            required: true,
        },
        race: {
            type: String,
            required: true,
        },
        age: {
            type: Number,
            required: true,
        },
        captureDate: {
            type: Date,
        },
    }
    , { timestamps: true }
);

module.exports = mongoose.model("Pet", petSchema);
